Add list itemLayoutAnimation documentation page and example (#6279)
software-mansion/react-native-reanimated: packages/docs-reanimated/docs/layout-animations/list-layout-animations.mdx

Reviewer: Can you change `itemLayoutAnimation` on the fly? i.e.

`itemLayoutAnimation = { condition ? first : second }`

Please include the answer in the page.

Author: Ok, I added this to the example in the app and will add info to docs. I think that I don't have to update the recording, though. Just mentioning this possibility in docs will be enough.

https://github.com/user-attachments/assets/1302fadd-4dd5-4640-a993-fae0f77963b6

Author: I added changes to the example app and code example to the **Remarks** section. I don't think I need to update the recording, information in the **Remarks** section should be enough.

https://github.com/user-attachments/assets/dfe6e148-a2d6-4545-8637-84431e75af7e

---
sidebar_position: 6
title: List Layout Animations
sidebar_label: List Layout Animations
---

> `itemLayoutAnimation` lets you define the animation that will be used when list items layout changes. You can use one of the predefined [layout transitions](/docs/layout-animations/layout-transitions#predefined-transitions) like `LinearTransition` or create your own [custom layout transition](/docs/layout-animations/custom-animations#custom-layout-transition).

Reviewer: ```suggestion
`itemLayoutAnimation` lets you define a [layout transition](/docs/layout-animations/layout-transitions) that's applied when list items layout changes. You can use one of the [predefined transitions](/docs/layout-animations/layout-transitions#predefined-transitions) like `LinearTransition` or create [your own transition](/docs/layout-animations/custom-animations#custom-layout-transition).
```

## Example

<Row>

<ThemedVideo
  sources={{
    light: '/recordings/layout-animations/listitem_light.mov',
    dark: '/recordings/layout-animations/listitem_dark.mov',
  }}
/>

<div style={{flexGrow: 1}}>

```jsx
import Animated, { LinearTransition } from 'react-native-reanimated';

function App() {
  return (
    <Animated.FlatList
      data={data}
      renderItem={renderItem}
      // highlight-next-line
      itemLayoutAnimation={LinearTransition}
    />
  );
}
```

</div>

</Row>

## Remarks

> - `itemLayoutAnimation` works only with a single-solumn `Animated.FlatList`,
> - it does not work when `numColumns` is greater than 1,

Reviewer: ```suggestion
- `itemLayoutAnimation` works only with a single-column `Animated.FlatList`, `numColumns` property cannot be grater than 1.
```

---
sidebar_position: 6
title: List Layout Animations
sidebar_label: List Layout Animations
---

`itemLayoutAnimation` lets you define a [layout transition](/docs/layout-animations/layout-transitions) that's applied when list items layout changes. You can use one of the [predefined transitions](/docs/layout-animations/layout-transitions#predefined-transitions) like `LinearTransition` or create [your own transition](/docs/layout-animations/custom-animations#custom-layout-transition).

## Example

<Row>

<ThemedVideo
  sources={{
    light: '/recordings/layout-animations/listitem_light.mov',
    dark: '/recordings/layout-animations/listitem_dark.mov',
  }}
/>

<div style={{flexGrow: 1}}>

```jsx
import Animated, { LinearTransition } from 'react-native-reanimated';

function App() {
  return (
    <Animated.FlatList
      data={data}
      renderItem={renderItem}
      // highlight-next-line
      itemLayoutAnimation={LinearTransition}
    />
  );
}
```

</div>

</Row>

## Remarks

- `itemLayoutAnimation` works only with a single-column `Animated.FlatList`, `numColumns` property cannot be grater than 1.

> - you can change the `itemLayoutAnimation` on the fly or disable it by setting it to `undefined`.

Reviewer: ```suggestion
- You can change the `itemLayoutAnimation` on the fly or disable it by setting it to `undefined`.
```
